.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,42 @@
+name: End-to-end tests
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+jobs:
+  e2e:
+    runs-on: ubuntu-latest
+    environment: e2e
+    env:
+      IONOS_TOKEN: ${{ secrets.IONOS_TOKEN }}
+      IONOSCLOUD_MACHINE_IMAGE_ID: ${{ vars.IONOSCLOUD_MACHINE_IMAGE_ID }}
+      CONTROL_PLANE_ENDPOINT_LOCATION: ${{ vars.CONTROL_PLANE_ENDPOINT_LOCATION }}
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Setup Go environment
+      uses: actions/setup-go@v5
+      with:
+        go-version: 1.22
+
+    - name: Install kubectl and kustomize
```

Comment:
There is no need for this step,

kubectl and kustomize should be available by builtin runners